Formula/spidermonkey.rb

```diff
@@ -20,49 +21,52 @@ class Spidermonkey < Formula
     sha256 cellar: :any, mojave:   "8c0b46bc04a7e95f99262969b22cc311ee1f7d83413af05865318743ccd96944"
   end
 
-  depends_on :macos # Due to Python 2
+  depends_on "autoconf@2.13" => :build
+  depends_on "pkg-config"    => :build
+  depends_on "python@3.9"    => :build
+  depends_on "rust"          => :build
+
+  depends_on "icu4c"
   depends_on "nspr"
-  depends_on "readline"
 
-  conflicts_with "narwhal", because: "both install a js binary"
+  uses_from_macos "libedit"
 
   def install
-    cd "js/src" do
-      # Remove the broken *(for anyone but FF) install_name
-      inreplace "config/rules.mk",
-        "-install_name @executable_path/$(SHARED_LIBRARY) ",
-        "-install_name #{lib}/$(SHARED_LIBRARY) "
-
-      # The ./configure script assumes that it can find readline
-      # just as "-lreadline", but we want it to look in opt/readline/lib
-      inreplace "configure", "-lreadline", "-L#{Formula["readline"].opt_lib} -lreadline"
-    end
+    # TODO: remove this `inreplace` statement in future because
+    # upstream has removed the `sdk_max_version`
+    inreplace "build/moz.configure/toolchain.configure",
+                'sdk_max_version = Version("11.1")',
+                'sdk_max_version = Version("11.99")'
 
-    # The ./configure script that comes with spidermonkey 1.8.5 makes some mistakes
-    # with Xcode 12's default setting of -Werror,implicit-function-declaration
-    ENV.append "CFLAGS", "-Wno-implicit-function-declaration"
+    inreplace "config/rules.mk",
+              "-install_name $(_LOADER_PATH)/$(SHARED_LIBRARY) ",
+              "-install_name #{lib}/$(SHARED_LIBRARY) "
```

Comment:
Are you sure we still need this? If it's using `$(_LOADER_PATH)` it might work for us now.

Reply:
This is copied from formular `gjs`. I'm not sure whether this formula needs this